metadata-service/plugin/src/main/java/com/datahub/plugins/loader/PluginPermissionManagerImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.datahub.plugins.loader;
+
+import com.datahub.plugins.common.PluginPermissionManager;
+import com.datahub.plugins.common.SecurityMode;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.security.CodeSource;
+import java.security.Permissions;
+import java.security.ProtectionDomain;
+import java.security.cert.Certificate;
+
+
+public class PluginPermissionManagerImpl implements PluginPermissionManager {
+
+  private SecurityMode _securityMode;
+  public PluginPermissionManagerImpl(SecurityMode securityMode) {
+    this._securityMode = securityMode;
+  }
+
+  @Override
+  public ProtectionDomain createProtectionDomain(Path sourceCodeDirectory) {
```

Comment:
What happens if the sourceCodeDirectory parameter is null ?

Reply:
It is never going to be null because it is the plugin-home directory